inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/eltwise/eltwise_kernel_b_fs_yx_fsv16.cpp

```diff
@@ -69,14 +93,37 @@ JitConstants EltwiseKernel_b_fs_yx_fsv16::MakeLoadJitConstants(const eltwise_par
                                                         "input" + std::to_string(input.index) +
                                                         "[0]"));
                     } else {
-                        std::string block_read_str = "BLOCK_READN(INPUT" + std::to_string(input.index) + "_TYPE, " +
-                                                                 "BLOCK_SIZE, " +
-                                                                 "input" + std::to_string(input.index) + ", " +
-                                                                 "INPUT" + std::to_string(input.index) + "_GET_INDEX(b, f_block*16, y, x))";
-                        jit.AddConstant(MakeJitConstant(name,
-                                                        "TO_TYPE(MAKE_VECTOR_TYPE(ACCUMULATOR_TYPE, BLOCK_SIZE), " + block_read_str + ")"));
+                        const std::string idx_order = "INPUT" + std::to_string(input.index) + "_IDX_ORDER";
+                        jit.AddConstant(MakeJitConstant(idx_order, "b, f_block*16, y, x"));
+
+                        bool feature_broadcasting = (params.inputs[input_idx].Feature().v == 1 && params.output.Feature().v != 1);
+
+                        const std::string block_read_str = "TO_TYPE(MAKE_VECTOR_TYPE(ACCUMULATOR_TYPE, BLOCK_SIZE), BLOCK_READN(INPUT" +
+                                                                std::to_string(input.index) + "_TYPE, BLOCK_SIZE, " +
+                                                                "input" + std::to_string(input.index) + ", " +
+                                                                "GET_INDEX(INPUT, " + std::to_string(input.index) + ", " + idx_order + ")))";
+                        if (feature_broadcasting) {
+                            const std::string broadcast_name = "DO_FEATURE_BROADCAST" + std::to_string(op_num);
+                            std::string sub_group_broadcast;
+                            if (GetBlockSize(params) == 1) {
+                                sub_group_broadcast = "\\\n\ttmp_b" + std::to_string(op_num) +
+                                                    " = sub_group_broadcast(tmp_b" + std::to_string(op_num) + ", 0);";
+                            } else {
+                                sub_group_broadcast = "\\\n\tunroll_for (uint i = 0; i < BLOCK_SIZE; ++i) tmp_b" + std::to_string(op_num) +
+                                                    "[i] = sub_group_broadcast(tmp_b" + std::to_string(op_num) + "[i], 0);";
+                            }
+
+                            std::string broadcast_value = "\\\n\tMAKE_VECTOR_TYPE(ACCUMULATOR_TYPE, BLOCK_SIZE) tmp_b" + std::to_string(op_num) +
```

Comment:
Don't you need to have different output type for BLOCK_SIZE==1? I assume that MAKE_VECTOR_TYPE(ACCUMULATOR_TYPE, 1) will return something like `float1` which is invalid

Reply:
data_types.cl defines below
#define MAKE_VECTOR_TYPE_IMPL_1(elem_type)  elem_type
..
#define MAKE_VECTOR_TYPE(elem_type, size)   CAT(MAKE_VECTOR_TYPE_IMPL_, size)(elem_type)

so I think it will convert float when BLOCK_SIZE is 1.